protocol/contracts/v2/access/TempleElevatedAccess.sol

```diff
@@ -0,0 +1,107 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (v2/access/TempleElevatedAccess.sol)
+
+import { ITempleElevatedAccess } from "contracts/interfaces/v2/access/ITempleElevatedAccess.sol";
+import { CommonEventsAndErrors } from "contracts/common/CommonEventsAndErrors.sol";
+
+/**
+ * @notice Inherit to add Executor and Rescuer roles for DAO elevated access.
+ */ 
+abstract contract TempleElevatedAccess is ITempleElevatedAccess {
+    /**
+     * @notice A set of addresses which are approved to execute emergency operations.
+     */ 
+    mapping(address => bool) public override rescuers;
+
+    /**
+     * @notice A set of addresses which are approved to execute normal operations on behalf of the DAO.
+     */ 
+    mapping(address => bool) public override executors;
+
+    /**
+     * @notice Explicit approval for an address to execute a function.
+     * allowedCaller => function selector => true/false
+     */
+    mapping(address => mapping(bytes4 => bool)) public override explicitFunctionAccess;
+
+    /**
+     * @notice Under normal circumstances, rescuers don't have access to admin/operational functions.
+     * However when rescue mode is enabled (by rescuers or executors), they claim the access rights.
+     */
+    bool public override inRescueMode;
+
+    constructor(address initialRescuer, address initialExecutor) {
+        rescuers[initialRescuer] = true;
+        executors[initialExecutor] = true;
+    }
+
+    /**
+     * @notice Set the contract into or out of rescue mode.
+     * Only the rescuers are allowed to set.
+     */
+    function setRescueMode(bool value) external override {
+        if (!rescuers[msg.sender]) revert CommonEventsAndErrors.InvalidAccess();
+        emit RescueModeSet(value);
+        inRescueMode = value;
+    }
+
+    /**
+     * @notice Grant `account` the emergency operations role
+     */
+    function setRescuer(address account, bool value) external override onlyElevatedAccess {
```

Comment:
I would use the set, accept format for these critical functions/roles. eg. like FRAX does `setAdmin`, `acceptAdmin`

Reply:
Ah yes, nice one will do

Reply:
Done: https://github.com/TempleDAO/temple/pull/814/commits/12985afec975531fc88e804859d7d6565742f903